### ISSUE TITLE
`copy-hint-url` does not copy the link that `follow-hint` would go to.

### ISSUE THREAD

**Issue by Zaijab** [atlas-engineer/nyxt#1956]
**Describe the bug**
The URL that `follow-hint` goes to and `copy-hint-url` copies is sometimes different. I hope I am correct in assuming that we wish for the two to be the same.
For most links, the two URLs are the same. However, on Youtube I notice that when I copy a URL from a hint, then I only get the URL "https://www.youtube.com/watch" rather than the expected form "https://www.youtube.com/watch?v=ltLUadnCyi0" which `follow-hint` would give (forgive the inadvertent advertising I wished to use an example).

In looking into the source code, both `follow-hint` and `copy-hint-url` use the input of `nyxt/web-mode:query-hints` to a lambda function.
`follow-hint` uses the car of the input to this lambda `(first nyxt/web-mode::results)` whereas `copy-hint-url` renders the URL by doing the following `(render-url (url (first nyxt/web-mode::results)))`.

Could this difference in the code be the cause of the difference in behaviors? I tried looking into the definition for `url` and `render-url` but my progress has been extremely slow due to my unfamiliarity with the codebase. 

**Precise recipe to reproduce the issue**
This is a website-specific issue I have not replicated with Epiphany / GNOME Web.
This happens to me on [Youtube.com]. 

1. Go to www.youtube.com
2. Run the function `copy-hint-url` on a video. (Notice that the URL shows in this menu)
3. Paste the copied text anywhere (I used `set-url` by hitting 'o' in Vi-normal mode)

Compare this to:

1. Go to www.youtube.com
2. Run the function `follow-hint` on a video.
3. You should appear on the right URL but it's different than the one given by `copy-hint-url`.

For website-specific issues:
Can you reproduce this issue with Epiphany / GNOME Web (https://wiki.gnome.org/Apps/Web)?

**Information**
- OS name+version: GNU Guix System 1.3.0
- Graphics card and driver:
```
zjabbar@tao ~$ sudo lshw -C video
  *-display                 
       physical id: 0
       bus info: pci@0000:01:00.0
       version: a1
       width: 64 bits
       clock: 33MHz
       capabilities: pm msi pciexpress bus_master cap_list rom
       configuration: driver=nouveau latency=0
       resources: irq:140 memory:53000000-53ffffff memory:40000000-4fffffff memory:50000000-51ffffff ioport:5000(size=128) memory:54000000-5407ffff
```
- Desktop environment / Window manager name+version: EXWM 0.26
- How you installed Nyxt (Guix pack, package manager, build from source): GUIX Pack
- Information from `show-system-information`:

```
Nyxt version: 2.2.3
Renderer version: GI-GTK
Operating system kernel: Linux 5.15.6-gnu
Lisp implementation: SBCL 2.1.9 (Dynamic space size: 3221225472)
Features: (:WEBKIT2 :WEBKIT2-2.34 :WEBKIT2-PASTE-PLAINTEXT :WEBKIT2-TRACKING
 :WEBKIT2-MUTE :WEBKIT2-EMOJI :WEBKIT2-MEDIA :WEBKIT2-SANDBOXING :GTK-3-22
 :GTK-3-20 :GTK-3-18 :GTK-3-16 :GTK-3-14 :GTK-3-12 :GTK-3-10 :GTK-3-8 :GTK-3-6
 :GTK-3-4 :GTK :GDK-3-22 :GDK-3-20 :GDK-3-18 :GDK-3-16 :GDK-3-14 :GDK-3-12
 :GDK-3-10 :GDK-3-8 :GDK-3-6 :GDK-3-4 :CAIRO-1-10 :CAIRO-1-12 :GDK-PIXBUF
 :GLIB-2-30 :GLIB-2-32 :GLIB-2-34 :GLIB-2-36 :GLIB-2-38 :GLIB-2-40 :GLIB-2-42
 :GLIB-2-44 :GLIB-2-46 :GLIB-2-48 :GLIB-2-50 :GLIB-2-52 :GLIB-2-54 :GLIB-2-56
 :GLIB-2-58 :GLIB :NYXT-2 :FSET-EXT-STRINGS :CUSTOM-HASH-TABLE-NATIVE :SWANK
 :PLUMP-UTF-32 :GLOBAL-VARS :DECLARE-TYPES :PARENSCRIPT :NAMED-READTABLES
 :LPARALLEL :21BIT-CHARS :CHUNGA :FLEXI-STREAMS :CLOSER-MOP :CL-PPCRE-UNICODE
 :CL-UNICODE :CL-PPCRE :BORDEAUX-THREADS :LPARALLEL.WITH-CLTL2
 :LPARALLEL.WITH-CAS :LPARALLEL.WITH-STEALING-SCHEDULER :SPLIT-SEQUENCE
 CHIPZ-SYSTEM:GRAY-STREAMS CFFI-FEATURES:FLAT-NAMESPACE CFFI-FEATURES:X86-64
 CFFI-FEATURES:UNIX :CFFI CFFI-SYS::FLAT-NAMESPACE ALEXANDRIA::SEQUENCE-EMPTYP
 :FAST-IO-SV :FAST-IO :SBCL-USES-SB-ROTATE-BYTE :CL-JSON-CLOS :CL-JSON
 :THREAD-SUPPORT :ASDF3.3 :ASDF3.2 :ASDF3.1 :ASDF3 :ASDF2 :ASDF :OS-UNIX
 :NON-BASE-CHARS-EXIST-P :ASDF-UNICODE :X86-64 :GENCGC :64-BIT :ANSI-CL
 :COMMON-LISP :ELF :IEEE-FLOATING-POINT :LINUX :LITTLE-ENDIAN
 :PACKAGE-LOCAL-NICKNAMES :SB-CORE-COMPRESSION :SB-LDB :SB-PACKAGE-LOCKS
 :SB-THREAD :SB-UNICODE :SBCL :UNIX)

ASDF version: 3.3.5
ASDF registries: (NYXT-SOURCE-REGISTRY ENVIRONMENT-SOURCE-REGISTRY USER-SOURCE-REGISTRY
                      USER-SOURCE-REGISTRY-DIRECTORY
                      DEFAULT-USER-SOURCE-REGISTRY SYSTEM-SOURCE-REGISTRY
                      SYSTEM-SOURCE-REGISTRY-DIRECTORY
                      DEFAULT-SYSTEM-SOURCE-REGISTRY)
Critical dependencies: (/gnu/store/w2fyhz3bbxi28m20rjxpd6974wpcvpmk-sbcl-cl-cffi-gtk-0.11.2-2.e9a46df/share/common-lisp/sbcl/cl-cffi-gtk/gtk/cl-cffi-gtk.asd
 /gnu/store/bdfnmkxaxsgq5a2073hgjsb700yhbjlb-sbcl-cl-gobject-introspection-0.3-1.d0136c8/share/common-lisp/sbcl/cl-gobject-introspection/cl-gobject-introspection.asd
 /gnu/store/vlx2jb76ihf3iwad5k67k47r7n3c7naa-sbcl-cl-webkit-3.4.0/share/common-lisp/sbcl/cl-webkit/webkit2/cl-webkit2.asd)

Guix version:
```
^ Guix version is blank for some reason.

  If you can't run `show-system-information`, try `nyxt --system-information` from
  a shell.  It this still does not work, please provide the following:
  - Nyxt version (from the `nyxt-version` command or `nyxt --version` from a shell):
  - Lisp implementation/version (if built from source):
  - Kernel name+version:
  - WebKitGTK+ | QtWebEngine version:

**Output when started from a shell**
```
zjabbar@tao ~$ nyxt
<INFO> [16:23:24] Listening to socket "/run/user/1000/nyxt/nyxt.socket".
Nyxt version 2.2.3
<INFO> [16:23:24] Loading Lisp file "/home/zjabbar/.config/nyxt/auto-config.lisp".
<INFO> [16:23:24] Loading Lisp file "/home/zjabbar/.config/nyxt/init.lisp".

; file: /home/zjabbar/.config/nyxt/init.lisp
; in: DEFINE-COMMAND HINT-MPV
;     (NYXT:DEFINE-COMMAND NYXT:HINT-MPV
;         (&KEY NYXT/WEB-MODE::ANNOTATE-VISIBLE-ONLY-P)
;       "Show a set of element hints, and copy the URL of the user inputted one.";       (NYXT/WEB-MODE:QUERY-HINTS "Copy element URL"
;                                  (LAMBDA (NYXT/WEB-MODE::RESULTS)
;                                    (NYXT::MPV (NYXT:RENDER-URL #)))))
; 
; caught STYLE-WARNING:
;   The variable NYXT/WEB-MODE::ANNOTATE-VISIBLE-ONLY-P is defined but never used.
; 
; compilation unit finished
;   caught 1 STYLE-WARNING condition
WARNING:
   redefining DEFAULT-MODES :AROUND (#<STANDARD-CLASS NYXT:USER-BUFFER>) in DEFMETHOD
WARNING:
   redefining DEFAULT-MODES :AROUND (#<STANDARD-CLASS NYXT:USER-PROMPT-BUFFER>) in DEFMETHOD
<INFO> [16:23:26] Loading history of 522 URLs from "/home/zjabbar/.local/share/nyxt/history/default.lisp".
<INFO> [16:23:33] Loading "https://www.youtube.com/feed/subscriptions".
<INFO> [16:23:36] Finished loading "https://www.youtube.com/feed/subscriptions".
<INFO> [16:24:41] Deleting socket "/run/user/1000/nyxt/nyxt.socket".
```

**Comment by aartaka** [comment]
It seems we weren't copying the URL query string when completing URLs. Thanks for reporting! Should be fixed in 9dfea80624d6595914b87fcf99efe2f7d4beedd5.

**Comment by aartaka** [comment]
Closing this as it should be fixed. Feel free to ping me if it isn't :)
